src/web/js/dashboard/Student.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import { Icon, Tooltip } from 'react-mdl';
+import { Card, CardTitle, CardText, Textfield, Button } from 'react-mdl';
+
+class Student extends Component {
+  state = {
+    id: this.props.details.id,
+    firstName: this.props.details.firstName,
+    lastName: this.props.details.lastName,
+    email: this.props.details.email,
+    editing: false
+  };
+  api = this.props.api;
+
+  handleClickEditStudent = () => {
+    this.setState({editing: true});
+  }
+
+  handleChange = (event) => {
+    this.setState({[event.target.id]: event.target.value});
+  }
+
+  handleClickRemoveStudent = () => {
+    if (confirm('Remove this student from the roster? This cannot be undone.')) {
+      this.props.snackBar('Student Removed. Please allow a few seconds for changes to appear.');
+      this.api.removeStudent(this.state.id).then(this.props.refreshParent);
+    }
+  }
+
+  handleSubmitEditStudent = (event) => {
+    event.preventDefault();
+    this.props.snackBar('Student Updated. Please allow a few seconds for changes to appear.');
```

Comment:
Gotta say I love the term `snackBar` even though I have no idea why it's called that.